gef.py

```diff
@@ -2172,8 +2172,13 @@ def read_cstring_from_memory(address, max_length=GEF_MAX_STRING_LENGTH, encoding
 
     char_t = cached_lookup_type("char")
     char_ptr = char_t.pointer()
-    res = gdb.Value(address).cast(char_ptr).string(encoding=encoding).strip()
-    res2 = res.replace('\n','\\n').replace('\r','\\r').replace('\t','\\t')
+    try:
+        res = gdb.Value(address).cast(char_ptr).string(encoding=encoding).strip()
+    except gdb.error:
+        length = min(address|(DEFAULT_PAGE_SIZE-1), max_length+1)
+        mem = bytes(read_memory(address, length)).decode("utf-8")
+        res = mem.split("\x00", 1)[0]
+    res2 = res.encode("unicode_escape")
```

Comment:
Have you tested this? My fear is that it won't work on python 3, actually, because this returns a `bytes` type.
If breaks things (and I think it will) then you can add `.decode("utf-8")` at the end, to make it a string again.

Reply:
This currently works for me in both py2/3

If I add `.decode("utf-8")` to the end it seems to break other things (non objc app) and always print addresses as strings:
<img width="667" alt="image" src="https://user-images.githubusercontent.com/1418260/46847477-3784c400-ce30-11e8-8747-993a85f8a77c.png">

Reply:
What do you mean it print addresses as string?
it looks good from your example.

Reply:
eg `$rip` is being rendered as a string when it shouldn't be

Reply:
got it.